dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/NativeHttpRestClient.java

```diff
@@ -0,0 +1,386 @@
+                CompletionSuggestionBuilder.SUGGESTION_NAME),
+                (parser, context) -> CompletionSuggestion.fromXContent(parser, (String)context)));
+        return entries;
+    }
+
+    public NativeHttpRestClient(final SSLContext sslContext,
+            final CredentialsProvider credentialsProvider, final String hostName, final int port) {
+
+        final String protocol = sslContext != null ? "https://" : "http://";
+
+        PoolingHttpClientConnectionManager cm;
+        SSLConnectionSocketFactory socketFactory = null;
+
+        if (sslContext != null) {
+            socketFactory = new SSLConnectionSocketFactory(sslContext);
+
+            Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create()
+                    .register("https", socketFactory).build();
+
+            cm = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+
+        } else {
+            cm = new PoolingHttpClientConnectionManager();
+        }
+        // Increase max total connection to 200
+        cm.setMaxTotal(200);
+        // Increase default max connection per route to 20
+        cm.setDefaultMaxPerRoute(20);
+        // Increase max connections for localhost:80 to 50
+        HttpHost localhost = new HttpHost(hostName, port);
+        cm.setMaxPerRoute(new HttpRoute(localhost), 50);
+
+        if (socketFactory != null) {
+
+            client = HttpClients.custom().setDefaultCredentialsProvider(credentialsProvider)
+                    .setRetryHandler(new MyRequestRetryHandler())
+                    .setSSLSocketFactory(socketFactory)
+                    .setConnectionManager(cm)
+                    .build();
+        } else {
+            client = HttpClients.custom().setDefaultCredentialsProvider(credentialsProvider)
+                    .setRetryHandler(new MyRequestRetryHandler())
+                    .setConnectionManager(cm)
+                    .build();
+        }
+        NativeHttpRestClient.url = protocol + hostName + ":" + port;
+
+        List<NamedXContentRegistry.Entry> namedXContentEntries = Collections.emptyList();
+
+        registry = new NamedXContentRegistry(
+                Stream.of(getDefaultNamedXContents().stream(), getProvidedNamedXContents().stream(),
+                        namedXContentEntries.stream())
+                        .flatMap(Function.identity()).collect(toList()));
+    }
+
+    /**
+     * Loads and returns the {@link NamedXContentRegistry.Entry} parsers provided by plugins.
+     */
+    static List<NamedXContentRegistry.Entry> getProvidedNamedXContents() {
+        List<NamedXContentRegistry.Entry> entries = new ArrayList<>();
+        for (NamedXContentProvider service : ServiceLoader.load(NamedXContentProvider.class)) {
+            entries.addAll(service.getNamedXContentParsers());
+        }
+        return entries;
+    }
+
+    public SearchResponse search(SearchRequest searchRequest, RequestOptions options) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'searchRequest' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5121287)